scheduler/file.c

```diff
@@ -325,21 +325,21 @@ cupsdRemoveFile(const char *filename)	/* I - File to remove */
   if (access(filename, 0))
     return (0);
 
-  cupsdLogMessage(CUPSD_LOG_DEBUG, "Securely removing \"%s\".", filename);
-
  /*
   * First open the file for writing in exclusive mode.
   */
 
   if ((fd = open(filename, O_WRONLY | O_EXCL)) < 0)
     return (-1);
 
+  cupsdLogMessage(CUPSD_LOG_DEBUG, "Securely removing \"%s\".", filename);
+
  /*
   * Delete the file now - it will still be around as long as the file is
   * open...
   */
 
-  if (unlink(filename))
+  if (unlinkat(fd, filename, 0))
```

Comment:
`unlinkat` is not portable enough for CUPS. Also, the file descriptor in this case is used for resolving relative paths and not for removing the currently open file.